torch/_dynamo/side_effects.py

```diff
@@ -150,8 +138,21 @@ def __contains__(self, item):
     def __getitem__(self, item):
         return self.id_to_variable[id(item)]
 
+    def check_allowed_side_effect(self, item):
+        from torch._dynamo.variables.misc import AutogradFunctionContextVariable
+
+        # People do things like self.dim = dim inside autograd.Function.
+        # These are benign.
+        if isinstance(item, AutogradFunctionContextVariable):
+            return True
```

Comment:
Alternatively, if this expands, a more dynamo-ish pattern would be

if `item.is_side_effect_safe()`

And AutogradFunctionContextVariable can override that to return True, everything else can invoke `is_side_effect_safe` 

Not important, fine as is for now.

Reply:
Agreed -- I'll leave it for now (because I hope more cases like this don't turn up)